lib/src/test/java/com/dynatrace/metric/util/MetricBuilderTest.java

```diff
@@ -300,25 +300,29 @@ public void testThrowsOnLineTooLong() {
       dimensions.add(Dimension.create(key, val));
     }
 
-    try {
-      Metric.builder("name")
-          .setPrefix("prefix")
-          .setLongCounterValueTotal(1)
-          .setDefaultDimensions(DimensionList.fromCollection(dimensions))
-          .serialize();
-    } catch (MetricException me) {
-      String expectedMessage =
-          "Line exceeds threshold of 2000 characters and cannot be ingested into Dynatrace";
-      // the serialized method will have the dimensions in a shuffled manner (due to the duplicate
-      // elimination using a map), so we assert that certain dimensions are included, but not the
-      // order.
-      assertTrue(me.getMessage().contains(expectedMessage));
-      assertTrue(me.getMessage().contains("dim0=val0"));
-      assertTrue(me.getMessage().contains("dim150=val150"));
-      assertTrue(me.getMessage().contains("dim249=val249"));
-      return;
-    }
-    // if the exception is not thrown this will make sure the test fails.
-    fail("should have thrown and returned from the catch block");
+    MetricException me =
+        assertThrows(
+            MetricException.class,
+            () ->
+                Metric.builder("name")
+                    .setPrefix("prefix")
+                    .setLongCounterValueTotal(1)
+                    .setDefaultDimensions(DimensionList.fromCollection(dimensions))
```

Comment:
You could put these steps separately above to ensure it's only thrown on `serialize()` and not earlier.